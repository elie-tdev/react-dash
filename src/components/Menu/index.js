export { default as MobileMenu } from './MobileMenu';
export { default as MobileMenuItem } from './MobileMenuItem';
