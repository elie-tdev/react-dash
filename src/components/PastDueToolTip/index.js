export { default as PastDueToolTip } from './PastDueToolTip';
