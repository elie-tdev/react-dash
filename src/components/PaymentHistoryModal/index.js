export { default as PaymentHistoryModal } from './PaymentHistoryModal';
