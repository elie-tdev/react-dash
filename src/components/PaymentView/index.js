export { default as PaymentView } from './PaymentView';
export { default as PaymentCurrent } from './PaymentCurrent';
export { default as PaymentPastDue } from './PaymentPastDue';
export { default as ScheduledPayment } from './ScheduledPayment';
export { default as NextPaymentView } from './NextPaymentView';
export { default as EnrollInAutoPay } from './EnrollInAutoPay';
export { default as NextPaymentCard } from './NextPaymentCard';
export { default as PastDueCard } from './PastDueCard';
export { default as AutoPayCard } from './AutoPayCard';
