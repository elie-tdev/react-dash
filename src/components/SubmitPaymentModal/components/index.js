export { default as CustomListItem } from './CustomListItem';
export { default as NewPaymentAccount } from './NewPaymentAccount';
export { default as EditPaymentAccount } from './EditPaymentAccount';
