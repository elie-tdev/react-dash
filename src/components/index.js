export * from './LoanGlance';
export * from './Button';
export * from './Header';
export * from './Footer';
export * from './LabelValueColumn';
export * from './LabelValueRow';
export * from './Loader';
export * from './CenteredScreenContent';
export * from './PaymentView';
export * from './Pagination';
export * from './Screen';
export * from './LoanSelect';
export * from './StatementGlance';
export * from './Menu';
export * from './PaymentHistoryModal';
export * from './PaymentStatus';
export * from './PaymentType';
export * from './SubmitPaymentModal';
export * from './GridContainer';
export * from './PastDueToolTip';
