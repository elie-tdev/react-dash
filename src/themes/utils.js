export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
