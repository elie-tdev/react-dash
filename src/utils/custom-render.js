import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
