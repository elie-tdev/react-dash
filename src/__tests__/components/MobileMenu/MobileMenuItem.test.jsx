import React from 'react';

import { render } from '@/utils/custom-render';
import { MobileMenuItem } from '@/components';

describe('MobileMenuItem', () => {
  test('renders', () => {
    const { container } = render(<MobileMenuItem />);
    expect(container).toMatchSnapshot();
  });

  test('renders with an active item name', () => {
    const { container } = render(<MobileMenuItem activeItemName />);
    expect(container).toMatchSnapshot();
  });
});
