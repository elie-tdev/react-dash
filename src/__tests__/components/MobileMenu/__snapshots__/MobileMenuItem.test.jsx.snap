// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenuItem renders 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-37 makeStyles-row-15 makeStyles-menuItem-1"
    data-testid="NavigationMenuItem-mobile"
  >
    <div
      class="MuiBox-root MuiBox-root-38 makeStyles-iconContainer-3 makeStyles-column-17 makeStyles-icon-6"
    />
    <p
      class="MuiTypography-root makeStyles-itemName-4 MuiTypography-body1"
    />
  </div>
</div>
`;

exports[`MobileMenuItem renders with an active item name 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-75 makeStyles-row-53 makeStyles-menuItem-39"
    data-testid="NavigationMenuItem-mobile"
  >
    <div
      class="MuiBox-root MuiBox-root-76 makeStyles-iconContainer-41 makeStyles-column-55 makeStyles-activeIcon-45"
    />
    <p
      class="MuiTypography-root makeStyles-itemName-42 makeStyles-activeItemName-43 MuiTypography-body1"
    />
  </div>
</div>
`;
